toffy/image_stitching.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+import math
+import natsort as ns
+import skimage.io as io
+
+from toffy import json_utils
+from ark.utils import data_utils, load_utils, io_utils, misc_utils
+from mibi_bin_tools.io_utils import remove_file_extensions
+
+
+def get_max_img_size(run_dir):
+    """Retrieves the maximum FOV image size listed in the fun file
+        Args:
+            run_dir (str): path to the run directory containing the run json files """
+
+    run_name = os.path.basename(run_dir)
+    run_file_path = os.path.join(run_dir, run_name + '.json')
+
+    # retrieve all pixel width dimensions of the fovs
+    run_data = json_utils.read_json_file(run_file_path)
+    img_sizes = []
+    for fov in run_data['fovs']:
+        img_sizes.append(fov.get('frameSizePixels')['width'])
+
+    # return largest
+    max_img_size = max(img_sizes)
+
+    return max_img_size
+
+
+def stitch_images(tiff_out_dir, run_dir, channels=None):
+    """Creates a new directory containing stitched channel images for the run
+        Args:
+            tiff_out_dir (str): path to the extracted images
+            run_dir (str): path to the run directory containing the run json files
+            channels (list): list of channels to produce stitched images for, None will do all """
+
+    # remove old images
+    stitched_dir = os.path.join(tiff_out_dir, 'stitched_images')
```

Comment:
This is how I set things up because I was the only one stitching images, so I could keep track of overwriting. However, this is hacky, so we should change to a directory named the same as the run_name that has the stitched images

Reply:
mentioned this in the reviewNB comment but currently the images are saved to a subdir `D:\\Extracted_Images\\run_name\\stitched_images` but I could I think I could edit the watcher to change it to a completely new directory like `D:\\Stitched_Images\run_name` if that's better?

Reply:
Nope, that's fine, can mirror the same logic in the individual notebook